addons/website_event/controllers/main.py

```diff
@@ -91,6 +113,20 @@ def events(self, page=1, slug_tags=None, **searches):
         events = event_details.get('results', Event)
         events = events[(page - 1) * step:page * step]
 
+        default_country = None
+        event_location = website.is_view_active('website_event.event_location')
+        if event_location:
+            include_online_events = website.is_view_active('website_event.event_location_include_online')
+            default_country = self._get_default_country_search(events, include_online_events)
```

Comment:
I don't believe this works with paging? 

"events" doesn't contain all events in the database, only the current page (let's say page 1 by default).
So if you have a visitor from Belgium but the event inside Belgium only appears in page 3, then it will not work.

I could be wrong, did not test as it takes too much time to test, but needs to be checked.